stylefunction.js

```diff
@@ -612,6 +612,10 @@ export default function(olLayer, glStyle, source, resolutions = defaultResolutio
           if (textPadding !== padding[0]) {
             padding[0] = padding[1] = padding[2] = padding[3] = textPadding;
           }
+          const textMaxAngle = getValue(layer, 'layout', 'text-max-angle', zoom, f);
+          if (textMaxAngle !== undefined && placement === 'line') {
```

Comment:
textMaxAngle should never be undefined here, so it would be more efficient to only call `getValue()` when `placement` is `'line'`.

Reply:
@ahocevar see commit: d54c15c.
Thanks for the review and fast response!
I noticed that `text-max-angle` can also be set to `line-center` so I have fixed the code accordingly. It looks a lot better now.